py_ocpi/credentials/v_2_2_1/api/cpo.py

```diff
@@ -1,10 +1,99 @@
-from fastapi import APIRouter
+import uuid
+
+import requests
+
+from fastapi import APIRouter, Depends, HTTPException, status as fastapistatus
+from pydantic import ValidationError
+
+from py_ocpi.core.schemas import OCPIResponse
+from py_ocpi.crud import get_crud
+from py_ocpi.adapter import get_adapter
+from py_ocpi.core import status
+from py_ocpi.core.enums import ModuleID
+from py_ocpi.credentials.v_2_2_1.schemas import Credentials
 
 router = APIRouter(
     prefix='/credentials',
 )
 
 
-@router.get("/")
-def get_credentials():
-    return "credentials app created!"
+@router.get("/", response_model=OCPIResponse)
+async def get_credentials(crud=Depends(get_crud), adapter=Depends(get_adapter)):
+    try:
+        data_list = await crud.list(ModuleID.CredentialsAndRegistrations)
+        credentials = []
+        for data in data_list:
+            credentials.append(adapter.credentials_adapter(data).dict())
+        return OCPIResponse(
+            data=credentials,
+            **status.OCPI_1000_GENERIC_SUCESS_CODE,
+        )
+    except ValidationError:
+        return OCPIResponse(
+            data=[],
+            **status.OCPI_3001_UNABLE_TO_USE_CLIENTS_API,
+        )
+
+
+@router.post("/", response_model=OCPIResponse)
+async def post_credentials(credentials: Credentials, crud=Depends(get_crud), adapter=Depends(get_adapter)):
+    try:
+        # Check if the client is already registered
+        credentials_client_token = credentials.token
+        server_cred = crud.get(ModuleID.CredentialsAndRegistrations, credentials.token)
+        if server_cred:
+            raise HTTPException(fastapistatus.HTTP_405_METHOD_NOT_ALLOWED, "Client has been already registered")
+
+        # Retrieve the versions and endpoints from the client
+        response_versions = requests.get(credentials.url, auth=credentials_client_token)
+        response_endpoints = requests.get(response_versions.json()['data']['url'], auth=credentials_client_token)
```

Comment:
use httpx instead of requests to make it async, and use headers for sending auth token